src/core/file_format/trp.cpp

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "trp.h"
+#ifdef _WIN64
+#include <Windows.h>
+#elif
+#include <limits.h>
+#include <unistd.h>
+#endif
+
+TRP::TRP() = default;
+TRP::~TRP() = default;
+
+void TRP::GetNPcommID(std::filesystem::path trophyPath, int fileNbr) {
+    std::filesystem::path trpPath = trophyPath / "sce_sys/npbind.dat";
+    Common::FS::IOFile npbindFile(trpPath, Common::FS::FileAccessMode::Read);
+    if (!npbindFile.IsOpen()) {
+        return;
+    }
+    for (int i = 0; i < fileNbr; i++) {
+        std::vector<u8> vec(12);
```

Comment:
```suggestion
        auto& vec = NPcommID.emplace_back(12);
```
This will avoid an extra copy and you can remove the push_back as well